benchalerts/clients.py

```diff
@@ -80,13 +152,25 @@ class GithubRepoClient(_BaseClient):
     Environment variables
     ---------------------
     GITHUB_API_TOKEN
-        A Github API token with ``repo`` access.
+        A Github API token with ``repo`` access. Only used for Personal Access Token
+        authentication.
+    GITHUB_APP_ID
+        The numeric GitHub App ID you can get from its settings page. Only used for
+        GitHub App authentication.
+    GITHUB_APP_PRIVATE_KEY
+        The full contents of the private key file downloaded from the App's settings
+        page. Only used for GitHub App authentication.
     """
 
     def __init__(self, repo: str, adapter: Optional[HTTPAdapter] = None):
         token = os.getenv("GITHUB_API_TOKEN")
         if not token:
-            fatal_and_log("Environment variable GITHUB_API_TOKEN not found")
+            log.info(
+                "Environment variable GITHUB_API_TOKEN not found. Attempting to "
+                "authenticate as a GitHub App."
+            )
```

Comment:
Should we flip the order here? It only matters in the corner case where somebody has all the env vars set, but I'd think we'd want to default to an app when available

Reply:
Yeah that probably makes more sense. The code will be slightly messier but that's fine.